content/How to work with Stripe.md

```diff
@@ -0,0 +1,133 @@
+
+- Add this package [Stripe](https://github.com/stripe/react-stripe-elements)
+- Configure your App to use the same api key as for rails (make sure it is the same, as you start moving between envs you may forget it). Remember that there is a testing key and a live one.
+
+Add an env file to store your keys
+https://gist.github.com/fedeagripa/5f99c824308ba899d6f7deb5fb85ff76
+
+Add your Stripe wrapper
+
+https://gist.github.com/fedeagripa/f2f64556a820abf9b377fa16f256d0d2
+
+## 3) START USING PAYMENTS WITH STRIPE
+
+### CREDIT CARDS
+
+##### REACT - DO YOURSELF A FAVOR AND USE THE EXISTING COMPONENT
+
+I'm not a fan of reinventing the wheel by any means, the design this components provide are more than enough for 99% of the apps you will be building. But if you insist, be prepared to spend 2 weeks dealing with details instead of 2 days.
+
+https://gist.github.com/fedeagripa/385cb4c4f9ccf66f833749349a41426a
+
+
+##### RAILS - DON'T TRY TO STORE ALL THE INFO (IT'S ILEGAL)
+
+You will tend to store more credit card info that you need. The only info that you need to store in your database (for basic usage) is:
+  - `customer_id` : Stripe customer identifier that you will store in your User for example
+  - `card_id` : Stripe card identifier
+
+The `token_id` you will get from your frontend is a short lived token that is only needed for an atomic operation.
+
+Add a `customer_id` field to your user (or Shop in next example).
+Add a `card_id` to your user (or Shop in next example).
+
+Now take this service example (Shopify page example):
+https://gist.github.com/fedeagripa/478e2c3a1e5eb018341740504fcb272f
+
+And this simple controller:
+
+https://gist.github.com/fedeagripa/29f02be2c8ab39560d06de47e13c7681
+
+And that's all! You can start charging your users now!
+All fraud detections and customer service actions can be managed directly from Stripe's dashboard.
+
+### SUBSCRIPTIONS
+To create a subscription you need to define it, then create a product in Stripe (this last one is really clear looking at the dashboard, so i'm not going to explain it)
+
+##### CREATING THE SUBSCRIPTION
+https://gist.github.com/fedeagripa/538df087957a2019e8444417347a32ac
+
+In that model you will store attribures like: `expires_at`, `type` or even `provider` if later you want to extend to other providers like PayPal or Apple Pay
+
+Finally to create them on Stripe is quite simple:
+
+https://gist.github.com/fedeagripa/d49a024299d2b8e315ff0861d32e9534
+
+### COUPONS
+Coupons are the abstract concept of `30% off` for example, when you apply that coupon to a user that's called a `discount`.
+So you should define some discounts on Stripe and store their ids in your database to apply them to users.
+There are two types of coupons `percentage` & `fixed amount`, and any of them can be one time only or have the capability to be applied multiple times. So when you try to apply a coupon to a subscription for example, remember that it can fail if you reached the maximum usage number.
```

Comment:
```suggestion
There are two types of coupons `percentage` & `fixed amount`, and any of them can be one time only or have the capability to be applied multiple times. So when you try to apply a coupon to a subscription, for example, remember that it can fail if you reached the maximum usage number.
```